api/core/television/television.sendCommand.js

```diff
@@ -0,0 +1,30 @@
+const Promise = require('bluebird');
+const queries = require('./television.queries.js');
+
+module.exports = function sendCommand(functionName, params) {
+    return getDeviceType(params)
+      .then((deviceType) => {
+
+          if(!gladys.modules[deviceType.service] || !gladys.modules[deviceType.service].television)
+                return Promise.reject(new Error(`television : Module ${deviceType.service} does not exist or does not handle television.`));
+         
+          if(typeof gladys.modules[deviceType.service].television[functionName] != 'function') {
+            console.log(`television : Module ${deviceType.service} does not have function ${functionName}`)
+            return Promise.reject();
+          } 
+
+          params.deviceType = deviceType;
+        
+          // calling television module
+          return gladys.modules[deviceType.service].television[functionName](params);
+        });
+};
+
+function getDeviceType(params){
+    if(params.device) return gladys.utils.sqlUnique(queries.getDeviceTypeByDeviceIdOrByDeviceTypeId, [params.device, params.device, null, null]);
```

Comment:
Why do we want to allow a "device" key to be either a device id or a devicetype id ? It's confusing!

Reply:
You right, are you agree if I change this 'device' by 'deviceId', because in this case we pass the device.id to the sendCommand method.